aea/skills/behaviours.py

```diff
@@ -115,3 +118,128 @@ def is_time_to_act(self) -> bool:
         """
         now = datetime.datetime.now()
         return now > self._start_at and (now - self._last_act_time).total_seconds() > self.tick_interval
+
+
+class SequenceBehaviour(CompositeBehaviour, ABC):
+    """This behaviour executes sub-behaviour serially."""
+
+    def __init__(self, behaviour_sequence: List[Behaviour], **kwargs):
+        """
+        Initialize the sequence behaviour.
+
+        :param behaviour_sequence: the sequence of behaviour.
+        :param kwargs:
+        """
+        super().__init__(**kwargs)
+
+        self._behaviour_sequence = behaviour_sequence
+        assert len(self._behaviour_sequence) > 0, "at least one behaviour."
+        self._index = 0
+
+    @property
+    def current_behaviour(self) -> Optional[Behaviour]:
+        """Get the current behaviour."""
+        return None if self._index >= len(self._behaviour_sequence) else self._behaviour_sequence[self._index]
+
+    def _increase_index_if_possible(self):
+        if self._index < len(self._behaviour_sequence):
+            self._index += 1
+
+    def act(self) -> None:
+        """Implement the behaviour."""
+        while not self.done() and self.current_behaviour.done():
+            self._increase_index_if_possible()
+
+        if not self.done() and not self.current_behaviour.done():
+            self.current_behaviour.act_wrapper()
+
+    def done(self) -> bool:
+        """Return True if the behaviour is terminated, False otherwise."""
+        return self._index >= len(self._behaviour_sequence)
+
+
+class State(SimpleBehaviour, ABC):
+    """A state of a FSMBehaviour is a OneShotBehaviour"""
```

Comment:
So should it inherit from OneShotBehaviour then?

Reply:
Not necessarily. For instance, it might be a `CyclicBehaviour`. The docstring is not correct. However, it is a good idea to restrict it to `OneShotBehaviour` as you suggested.